packages/plugin-stage/sources/drivers/GitDriver.ts

```diff
@@ -1,19 +1,118 @@
 import {execUtils}     from '@berry/core';
-import {NodeFS}        from '@berry/fslib';
+import {NodeFS, xfs}   from '@berry/fslib';
 import {posix}         from 'path';
 
 import * as stageUtils from '../stageUtils';
 
 const MESSAGE_MARKER = `Commit generated via \`yarn stage\``;
 const COMMIT_DEPTH = 11;
 
-async function genCommitMessage(cwd: string) {
+
+async function genCommitMessage(cwd: string, changes: Array<string>, cachedFileLists: string) {
+
+  const updates = new Map();
+  const removes = new Map();
+  const adds = new Map();
+  const createdPackages = [];
+  const removedPackages = [];
+
+
+  for(const change of changes) {
+    const localPath = NodeFS.fromPortablePath(change);
+    const relativePath = posix.relative(cwd, localPath);
+    const doesExistCurrent = xfs.existsSync(localPath);
+    const doesExistPrev = cachedFileLists.indexOf(relativePath) !== -1;
+    if (doesExistCurrent && doesExistPrev) {
+      const {stdout:prevStdout} = await execUtils.execvp(`git`, [`show`, `HEAD~1:${relativePath}`], {cwd, strict: true});
+      const {stdout:currStdout} = await execUtils.execvp(`cat`, [`${localPath}`], {cwd, strict: true});
```

Comment:
In this case it's a bit different because while we can use `await Manifest.fromFile` for the current state, we can't do it for the previous one. So instead I suggest we simply add a `Manifest.fromText` function that operates the same way but on a text buffer. This way we can just send it the return value of `git show` and keep the same logic.